sdk/internal/recording/recording.go

```diff
@@ -433,3 +438,221 @@ var modeMap = map[RecordMode]recorder.Mode{
+	if err != nil {
+		return err
+	}
+	recordingId = resp.Header.Get(IdHeader)
+	return nil
+}
+
+func StopRecording(t *testing.T, options *RecordingOptions) error {
+	defer resetRecordingId()
+
+	if options == nil {
+		options = defaultOptions()
+	}
+
+	url := fmt.Sprintf("%v/%v/stop", options.HostScheme(), recordMode)
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		return err
+	}
+	if recordingId == "" {
+		return errors.New("Recording ID was never set. Did you call StartRecording?")
+	}
+	req.Header.Set("x-recording-id", recordingId)
+	_, err = client.Do(req)
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+	return nil
+}
+
+func AddUriSanitizer(replacement, regex string, options *RecordingOptions) error {
+	if options == nil {
+		options = defaultOptions()
+	}
+	url := fmt.Sprintf("%v/Admin/AddSanitizer", options.HostScheme())
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("x-abstraction-identifier", "UriRegexSanitizer")
+	bodyContent := map[string]string{
+		"value": replacement,
+		"regex": regex,
+	}
+	marshalled, err := json.Marshal(bodyContent)
+	if err != nil {
+		return err
+	}
+	req.Body = ioutil.NopCloser(bytes.NewReader(marshalled))
+	req.ContentLength = int64(len(marshalled))
+	_, err = client.Do(req)
+	return err
+}
+
+func (o *RecordingOptions) Init() {
+	if o.UseHTTPS {
+		o.Host = baseProxyURLSecure
+		o.Scheme = "https"
+	} else {
+		o.Host = baseProxyURL
+		o.Scheme = "http"
+	}
+}
+
+// This looks up an environment variable and if it is not found, returns the recordedValue
+func GetEnvVariable(t *testing.T, varName string, recordedValue string) string {
+	val, ok := os.LookupEnv(varName)
+	if !ok {
+		t.Logf("Could not find environment variable: %v", varName)
+		return recordedValue
+	}
+	return val
+}
+
+func LiveOnly(t *testing.T) {
+	if GetRecordMode() != modeRecording {
+		t.Skip("Live Test Only")
+	}
+}
+
+// Function for sleeping during a test for `duration` seconds. This method will only execute when
+// AZURE_RECORD_MODE = "record", if a test is running in playback this will be a noop.
+func Sleep(duration int) {
+	if GetRecordMode() == modeRecording {
+		time.Sleep(time.Duration(duration) * time.Second)
+	}
+}
+
+func GetRecordingId() string {
+	return recordingId
+}
+
+func GetRecordMode() string {
+	return recordMode
+}
+
+func InPlayback() bool {
+	return GetRecordMode() == modePlayback
+}
+
+func InRecord() bool {
+	return GetRecordMode() == modeRecording
+}
+
+func getRootCas() (*x509.CertPool, error) {
+	localFile, ok := os.LookupEnv("PROXY_CERT")
+
+	rootCAs, err := x509.SystemCertPool()
+	if err != nil {
+		rootCAs = x509.NewCertPool()
+	}
+
+	if !ok {
+		fmt.Println("Could not find path to proxy certificate, set the environment variable 'PROXY_CERT' to the location of your certificate")
+		return rootCAs, nil
+	}
+
+	cert, err := ioutil.ReadFile(localFile)
+	if err != nil {
+		fmt.Println("error opening cert file")
```

Comment:
Do we need to fmt.Println here? Would fmt.Errorf() be more appropriate? (also, including the file name could be helpful in diagnosing errors).

Reply:
All of this information is included in the error from `ioutil` which covers the error case nicely.